src/server.rs

```diff
@@ -0,0 +1,294 @@
+use std::env;
+use fresh_eyes::{
+    extract_pr_details, get_pull_request_reviews, Branch as LibBranch,
+    ForkRequest as LibForkRequest, PullRequest as LibPullRequest,
+};
+use fresheyes::git_hub_service_server::{GitHubService, GitHubServiceServer};
+use fresheyes::{Branch, ForkRequest, ForkResult, PrResponse, PullRequest, PullRequestDetails};
+use tonic::{transport::Server, Request, Response, Status};
+use dotenv::dotenv;
+
+pub mod fresheyes {
+    tonic::include_proto!("fresheyes");
+}
+
+#[derive(Debug, Default)]
+pub struct GitHubServiceImpl {
+    github_token: String,
+}
+
+impl GitHubServiceImpl {
+    pub fn new(github_token: String) -> Self {
+        Self { github_token }
+    }
+}
+#[tonic::async_trait]
+impl GitHubService for GitHubServiceImpl {
+    // Implementation of the fork_repository method
+    async fn fork_repository(
+        &self,
+        request: Request<ForkRequest>,
+    ) -> Result<Response<ForkResult>, Status> {
+        let fork_request = request.into_inner();
+
+        let fork_request = LibForkRequest {
+            owner: &fork_request.owner.clone(),
+            repo: &fork_request.repo.clone(),
+        };
+
+        // Call the fork method to fork the repository on GitHub
+        match fork_request.fork().await {
+            Ok(data) => {
+                // If the fork was successful, create a ForkResult instance
+                let fork_result = ForkResult {
```

Comment:
No need to recreate the `ForkResult` struct. data is already a `ForkResult`

Reply:
no, we still need enforce struct literal in this block. It will raise a compile error if not added

Reply:
In that case, to reduce the code you can do:
```rust
Ok(Response::new(ForkResult {
                    owner: data.owner,
                    repo: data.repo,
                    forked_repo: data.forked_repo,
                }))
 ```